src/tracker/tracker-funcs.jl

```diff
@@ -498,6 +501,134 @@ function addfloemasks!(props::Vector{DataFrame}, imgs::Vector{<:FloeLabelsImage}
     return nothing
 end
 
+"""
+    get_unmatched(props, matched)
+
+Return the floes in `props` that are not in `matched`.
+"""
+function get_unmatched(props, matched)
+    _on = mapreduce(df -> Set(names(df)), intersect, [props, matched]) |> collect
+    unmatched = antijoin(props, matched, on=_on)
+
+    # Add missing columns for joining
+    add_missing = ["area_mismatch", "corr"]
+    [unmatched[!, n] = [missing for _ in 1:nrow(unmatched)] for n in add_missing]
+
+    return unmatched
+end
+
+"""
+    get_trajectory_heads(pairs)
+
+Return the last row (most recent member) of each group (trajectory) in `pairs` as a dataframe.
+
+This is used for getting the initial floe properties for the next day in search for new pairs.
+"""
+function get_trajectory_heads(pairs::T) where {T<:AbstractDataFrame}
+    gdf = groupby(pairs, :uuid)
+    return combine(gdf, last)[:, names(pairs)]
+end
+
+"""
+    _swap_last_values!(df)
+
+Swap the last two values of the `area_mismatch` and `corr` columns for each group in `df`. For bookkeeping purposes for goodness of fit data during the tracking process.
```

Comment:
Say that at step `k` you have the following trajectories:
```julia
julia> trajectories[:, [:ID, :passtime, :area_mismatch, :corr]]
6×4 DataFrame
 Row │ ID     passtime             area_mismatch  corr           
     │ Int64  DateTime             Float64?       Float64?       
─────┼───────────────────────────────────────────────────────────
   1 │     1  2022-09-14T12:00:00            0.0        0.991
   2 │     1  2022-09-15T12:00:00      missing    missing        
   3 │     2  2022-09-14T12:00:00            0.0        0.988
   4 │     2  2022-09-15T12:00:00      missing    missing        

```
Here the values of `:area_mismatch`  and `:corr` in row `i` indicate the _goodness_ of the matching between the floe data in that row and those in the `i+1`th row of that trajectory.

Assume further that at step `k+1` the algorithm suggests those trajectories should be extended with the following matches:
```julia
julia> toextend[:, [:ID, :passtime, :area_mismatch, :corr]]
6×4 DataFrame
 Row │ ID     passtime             area_mismatch  corr           
     │ Int64  DateTime             Float64?       Float64?       
─────┼───────────────────────────────────────────────────────────
   1 │     1  2022-09-16T12:00:00            0.0        0.995
   2 │     2  2022-09-16T13:00:00            0.0        0.988
    
```

How to include `toextend` within `trajectories` while preserving the meaning of `:area_mismatch`  and `:corr`? The simplest way I could think of was to just include the new rows in `toextend` within `trajectories` and sort by `ID` and then by `passtime`. But this would mess up the goodness columns.
```julia
julia> trajectories[1:8, [:ID, :passtime, :area_mismatch, :corr]]
6×4 DataFrame
 Row │ ID     passtime             area_mismatch  corr           
     │ Int64  DateTime             Float64?       Float64?       
─────┼───────────────────────────────────────────────────────────
   1 │     1  2022-09-14T12:00:00            0.0        0.991
   2 │     1  2022-09-15T12:00:00      missing    missing      
   3 │     1  2022-09-16T12:00:00            0.0        0.995  
   4 │     2  2022-09-14T12:00:00            0.0        0.988
   5 │     2  2022-09-15T12:00:00      missing    missing
   6 │     2  2022-09-16T13:00:00            0.0        0.988           

```
A call to `_swap_last_values!` fixes that distortion.

Reply:
So how would the table trajectories look afterwards?

like this?

```julia
julia> trajectories[:, [:ID, :passtime, :area_mismatch, :corr]]
6×4 DataFrame
 Row │ ID     passtime             area_mismatch  corr           
     │ Int64  DateTime             Float64?       Float64?       
─────┼───────────────────────────────────────────────────────────
   1 │     1  2022-09-14T12:00:00            0.0        0.991
   2 │     1  2022-09-16T12:00:00            0.0        0.995  
   3 │     1  2022-09-15T12:00:00      missing    missing      
   4 │     2  2022-09-14T12:00:00            0.0        0.988
   5 │     2  2022-09-16T13:00:00            0.0        0.988           
   6 │     2  2022-09-15T12:00:00      missing    missing

```

Could you alternatively change the meaning of `area_mismatch` and `corr` so that they mean the goodness of matching between the floe data in row `i` and the newest available earlier row `i-n`? Would that break other things?

Reply:
I think @danielmwatkins was fine with the current interpretation of those columns.